### ISSUE TITLE
Feature request: enable custome names for all k2 images

### ISSUE THREAD

**Issue by Joomlalife** [joomlaworks/k2-v3-dev-build#25]
To my knowledge image names in the path are an important part of SEO optimazation. In 
fact I have customer sites with a high number of page impressions based on image names only. 

Therefor I suggest the feature to have custom (or original) names for all k2 images (items, users, categories) e.g. like picturename_catid.jpg.

Code suggestion can be found here: http://www.gavick.com/forums/general-discussion/seo-k2-image-18208.html

This request is for k2 v3 as well as for k2 v2.x - let me know if I should post it also there.

Please let us know if you consider this request and if not if it could be done with a plugin.

**Comment by Robdebert** [comment]
This is a looong wanted feature. Is it planned to be integrated? Perhaps with a switch in the SEO-Options Panel "Use old md5-filenames or seo-filenames?" so backward-compatibility is provided.

**Comment by heyjoecampbell** [comment]
Good point @Robdebert - backward compatibility would be optimal

**Comment by fevangelou** [comment]
The impact to SEO is close to zero because of the use of the alt html attribute. We have discussed this extensively in the forum in the past. If your images are not being crawled by Google, check your robots.txt file.

**Comment by simbus82** [comment]
@fevangelou , sorry, you are veeeeery wrong.

Google Images for example check a correspondebce between file name and alt tag, and give to image more "power" related to the search term.
The semantic engine see file name, alt tag, anchor text and correspondence with page Headers and Title

Google Official
https://support.google.com/webmasters/answer/114016?hl=en

Neil Patel (you must know who is!)
http://www.quicksprout.com/2012/11/05/image-optimization-102-the-advance-guide-to-seo-for-images/

Shopify blog
http://www.shopify.com/blog/7412852-10-must-know-image-optimization-tips

Yoast
https://yoast.com/image-seo/

And every SEO say that file name is important.

Leave the file name untouched with only the add of your so loved uniqueId and a tag for image size, is a practice that have NO COST in writing code and do a lot of good work for SEO.

**Comment by UberGruber** [comment]
+1 to this feature and to @simbus82 statement. It's totally necessary to either to keep the name of the uploaded image adding the k2 item id for uniqueness, or to allow custom names on upload. 
It's been a while now that robot.txt [has changed](https://github.com/joomla/joomla-cms/blob/staging/robots.txt.dist).

**Comment by Joomlalife** [comment]
hi all and  

as v3 got "stuck" and I had other projects I did not immediately follow up on your points.

@kavadas: I modified the robot.txt file and about 1,5 years later I can only confirm the drastic drop in ranking. Changes in Google rankings aside, I can over long time observe that

a) pictures I insert via the editor with alt tag and the same name in \images\ rank accordingly. As it is a site current events I can also observe that they fall back over time which gives more weight to images in categories and subcategories as they stay rather stable (and I can compare with competitor's sites.

b) images on the old site (prior k2 & Joomla) that have been part of the site structure in categories and articles where shown as image results with any google search. Today they are not displayed any longer. It is a very small niche and we use the same picture in the new site. Thus there is a pretty good and clear argument for causality.

Therefor unlike @fevangelou, I think there is a clear case for SEO impact.

I see the same for PDFs that we attached to k2 items which we also moved into one designated folder. 

I therefore think it is still a great feature request, I will open new if not done with this ticket.
